### DISCUSSION TITLE
Importing replays on Android fails [Model creation failed]

### DISCUSSION THREAD

**Discussion by counter185** [ppy/osu#15511] [Q&A] [answered]
Don't know if this is the right place to report a bug, but I can't open issues so might as well try here. A commit was recently pushed that added the ".osr" intent filter to the main game activity, unfortunately importing replays always fails
This is the error:
![Screenshot_2021-11-07-07-25-05-949_sh ppy osulazer](https://user-images.githubusercontent.com/33550839/140644472-f03cfd8d-01d3-499b-88fe-c783faf43b7a.jpg)

and the exact same file imports just fine on PC:
![import_pc](https://user-images.githubusercontent.com/33550839/140644479-e154934d-eb75-4e32-97f6-c8959fc44f2b.png)

Log:
[database.log](https://github.com/ppy/osu/files/7492212/database.log)

From what I can tell, on Android the game is trying to read the `osr` file as an archive, which it isn't
I tried fixing this on my own but got lost in what code does what after 5 minutes and gave up

Phone: Xiaomi Mi 11 Lite 5G, Snapdragon 780G, Android 11

File managers I tried (all of which have the same issue):
- Xiaomi's built-in file manager - that one I think sends the wrong mime-type every time, which means it can't even import beatmaps
- X-plore - this one can import beatmaps properly
- Android 10's built-in file manager - same as above

How to reproduce:
1. Get any replay file (in my case it was a replay from the current Stable version, but these work with Lazer PC, so they should also work here)
2. Use a file manager to open the file, when asked what app to use, choose osu! obviously.
3. You will get the same error as in the first screenshot

**Comment by peppy** [comment, marked as answer]
Will be resolved via https://github.com/ppy/osu/pull/15591.
